activesupport/lib/active_support/tagged_logging.rb

```diff
@@ -57,8 +57,15 @@ def tags_text
     end
 
     def self.new(logger)
-      # Ensure we set a default formatter so we aren't extending nil!
-      logger.formatter ||= ActiveSupport::Logger::SimpleFormatter.new
+      logger = logger.dup
```

Comment:
Any special reason `#dup` was used here? I have opened https://github.com/rails/rails/pull/40759 to address a side-effect of not using `#clone`.